src/odemis/driver/ueye.py

```diff
@@ -678,8 +715,27 @@ def __init__(self, name, role, device=None, **kwargs):
 
             self._metadata[model.MD_DET_TYPE] = model.MD_DT_INTEGRATING
 
-            res = (sensorinfo.nMaxWidth, sensorinfo.nMaxHeight)
-            self._metadata[model.MD_SENSOR_SIZE] = self._transposeSizeToUser(res)
+            self._sensor_res = (sensorinfo.nMaxWidth, sensorinfo.nMaxHeight)
+            sensor_res_user = self._transposeSizeToUser(self._sensor_res)
+
+            if max_res is None:
+                max_res = self._sensor_res
+            else:
+                max_res = tuple(max_res)
+            if not all(1 <= mr <= r for mr, r in zip(max_res, sensor_res_user)):
+                raise ValueError(f"max_res has to be between (1, 1) and {sensor_res_user}, but got {max_res}")
+            max_res_hw = self._transposeSizeFromUser(max_res)
+
+            # compare the max_res parameter and the resolution of the camera sensor
+            # if the max_res param is smaller image cropping will be set through AOI
+            rect_aoi = IS_RECT()
+            rect_aoi.s32X = (self._sensor_res[0] - max_res_hw[0]) // 2
+            rect_aoi.s32Y = (self._sensor_res[1] - max_res_hw[1]) // 2
+            rect_aoi.s32Width = max_res_hw[0]
+            rect_aoi.s32Height = max_res_hw[1]
+            self.SetAOI(rect_aoi)
```

Comment:
Can you place this in a `if` block to only do it if it's needed. I'm a little afraid to break cases if SetAOI() raises an error even if passing the full resolution. (something like `if self._sensor_res != max_res_hw`).